xarray/convert.py

```diff
@@ -157,34 +161,48 @@ def _iris_cell_methods_to_str(cell_methods_obj):
     return ' '.join(cell_methods)
 
 
+def _name(iris_obj, default='unknown'):
+    """ Mimicks `iris_obj.name()` but with different name resolution order.
+
+    Same as iris_obj.name() method, but using iris_obj.var_name first to enable
+    roundtripping.
+    """
+    return iris_obj.var_name or iris_obj.standard_name or \
+        iris_obj.long_name or str(iris_obj.attributes.get('STASH', '')) or \
```

Comment:
is `iris_obj.attributes.get('STASH', '')` a normal Iris thing? That looks a little weird to me.

Reply:
For simplicity, I copied the code of Iris' [`.name()`](https://github.com/SciTools/iris/blob/master/lib/iris/_cube_coord_common.py#L87) method, and put `var_name` first to allow round-tripping DataArray -> Cube -> DataArray.

STASH is a variable identifier of the native file format of the Met Office's models. It is in `name()` so that model output for which Iris cannot find a standard name is not all shown as 'unknown'.

I could remove `iris_obj.attributes.get('STASH', '')`.

Reply:
Yes, I think my preference would be to remove `'STASH'`. It feels like an undocumented internal detail of Iris.